libcontainer/init_linux.go

```diff
@@ -140,6 +142,30 @@ func finalizeNamespace(config *initConfig) error {
 	return nil
 }
 
+// lateCgroups applies any cgroup limits which must be done as late as possible.
+// Specifically, this refers to pids.max which runs into issues when Go likes to
+// start spawning new OS threads seemingly at random.
+func lateCgroups(config *initConfig) error {
+	if path, ok := config.CgroupPaths["pids"]; ok {
+		// XXX: We have to bootstrap this here because the manager hasn't joined
```

Comment:
We probably need additional synchronization point between parent and child process to make this predictable.